src/components/assignment/lecture.tsx

```diff
@@ -393,11 +411,23 @@ export const LectureComponent = () => {
           />
         ) : null}
       </Typography>
-      <Stack>
+      <Stack direction={'row'} spacing={2}>
         <Typography variant={'h6'}>Assignments</Typography>
+        <GroupsDropDownMenu
+          assignments={assignments}
+          chosenGroup={chosenGroup}
+          setChosenGroup={setChosenGroup}
+        />
       </Stack>
 
-      <AssignmentTable lecture={lecture} rows={assignmentsState} />
+      <AssignmentTable
+        lecture={lecture}
+        rows={
+          chosenGroup === 'All'
```

Comment:
I'm not really a fan of having this hard-coded "All" as a special case that we have to handle separately in a few places in the code... We can't get rid of it. But maybe it would make sense to introduce a constant, say, `ALL_GROUPS_VALUE`?
(I'm not going to insist on doing this though. Just a suggestion to think about.)